sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -65,7 +66,7 @@ def _weight_func(dist):
     # Dist could be multidimensional, flatten it so all values
     # can be looped
     with np.errstate(divide='ignore'):
-        retval = 1. / dist
+        retval = 1. / (dist + 1e-8)
```

Comment:
I'm not sure about this fix. I think we still want to check that it works with a weight function which can be inf.

Reply:
1e-8 should only added to the zero entries otherwise or just avoid the division by zero no?

Reply:
The context manager just above and the comment tells me that the writers of this code were aware of this division by zero and wanted to keep testing against this specific behavior. The issue is that there's another warning not coming from the division but from an operation on an inifinite weight. That's why I'd rather keep it this way and suppress the warning in the test